src/rtracer.js

```diff
@@ -221,7 +221,11 @@ const hapiPlugin = ({
 
     if (echoHeader) {
       server.ext('onPreResponse', async (request, h) => {
-        request.response.header(headerName, id())
+        if (request.response.output) { // Response is a Boom error
```

Comment:
It seemed weird to check the "isBoom" property but then use the "output" property so I stuck with one but added a comment